railties/lib/rails/commands/commands_tasks.rb

```diff
@@ -103,6 +110,18 @@ def new
       end
     end
 
+    def rake
+      command = parse_command(ARGV.shift)
+
+      case command
+        when 'help' then ARGV.unshift('--help')
+        when 'version' then ARGV.unshift('--version')
+        else  ARGV.unshift(command)
+      end
```

Comment:
`parse_command` turns `--help` into `help` and this undoes it. We should find a way to not run the command through `parse_command`.